src/icesat2_tracks/analysis_db/B05_define_angle.py

```diff
@@ -374,143 +204,372 @@ def linear(self, radial_axis="period", ax=None, cbar_flag=True):
         self.ax = ax
 
 
-font_for_print()
-F = M.figure_axis_xy(6, 5.5, view_scale=0.7, container=True)
-gs = GridSpec(8, 6, wspace=0.1, hspace=3.1)
-color_schemes.colormaps2(21)
+def define_angle(
+    track_name: str = Option(..., callback=validate_track_name_steps_gt_1),
+    batch_key: str = Option(..., callback=validate_batch_key),
+    ID_flag: bool = True,
+    output_dir: str = Option(..., callback=validate_output_dir),
+    verbose: bool = False,
+):
+    """
+    TODO: add docstring
+    """
 
-cmap_spec = plt.cm.ocean_r
-clev_spec = np.linspace(-8, -1, 21) * 10
+    track_name, batch_key, test_flag = init_from_input(
+        [
+            None,
+            track_name,
+            batch_key,
+            ID_flag,
+        ]
+    )
 
-cmap_angle = color_schemes.cascade_r
-clev_angle = np.linspace(0, 4, 21)
+    hemis, batch = batch_key.split("_")
```

Comment:
The second retuned values doesnt seem to be used in the rest of the code
```suggestion
    hemis, _= batch_key.split("_")
```